ci-operator/step-registry/aws/c2s/init-token-service/aws-c2s-init-token-service-commands.sh

```diff
@@ -0,0 +1,326 @@
+        return 2
+    fi
+    pushd ${path} || return 1
+    for i in *.yaml; do
+        match_count=$(grep -c "${matched}" "${i}")
+        if [ $match_count -ne 0 ]; then
+            echo "[WARN] Remove CredentialsRequest ${i} which is a ${matched} CR"
+            rm -f ${i}
+            [ $? -ne 0 ] && echo "[ERROR] error remove CredentialsRequest ${i}" && return 1
+        fi
+    done
+    popd || return 1
+    return 0
+}
+
+
+# ------------------------------
+# create secret files
+# ------------------------------
+
+cr_yaml_d=$(mktemp -d)
+echo "extracting CR from image $RELEASE_IMAGE_LATEST"
+
+oc adm release extract ${RELEASE_IMAGE_LATEST} --credentials-requests --cloud=aws --to "${cr_yaml_d}" || exit 1
+
+echo "Extracted CR files:"
+ls $cr_yaml_d
+
+annotation="TechPreviewNoUpgrade"
+remove_tech_preview_feature_from_manifests ${cr_yaml_d} ${annotation} || exit 1
+
+credentials_requests_files=`mktemp`
+ls ${cr_yaml_d} > ${credentials_requests_files}
+
+echo "CRs to be processed:"
+cat "${credentials_requests_files}"
+
+while IFS= read -r item
+do
+    name=$(yq-go r "${cr_yaml_d}/${item}" 'spec.secretRef.name')
+    ns=$(yq-go r "${cr_yaml_d}/${item}" 'spec.secretRef.namespace')
+    echo "creating secret file for: \"${ns}\" \"${name}\""
+    create_secret_file ${ns} ${name} ${temp_cred_provider_role} ${key_id} ${key_sec}
+done < ${credentials_requests_files}
+
+# ------------------------------
+# set up token refresh service
+# ------------------------------
+
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-namespace.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cap-token-refresh
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-role.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cap-token-refresh
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - patch
+- apiGroups:
+  - cloudcredential.openshift.io
+  resources:
+  - credentialsrequests
+  verbs:
+  - list
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-rolebinding.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cap-token-refresh
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cap-token-refresh
+subjects:
+- kind: ServiceAccount
+  name: cap-token-refresh
+  namespace: cap-token-refresh
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-serviceaccount.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cap-token-refresh
+  namespace: cap-token-refresh
+EOF
+
+
+ca_file=`mktemp`
+cat "${CLUSTER_PROFILE_DIR}/shift-ca-chain.cert.pem" > ${ca_file}
+cat "/var/run/vault/mirror-registry/client_ca.crt" >> ${ca_file}
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-certs-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cap-certs
+  namespace: cap-token-refresh
+type: Opaque
+stringData:
+  cap.pem: |
+`echo -n ${temp_cred_provider_cert_b64} | base64 -d | sed -e 's/^/    /'`
```

Comment:
Its best not to template a yaml. Instead you may want to save a stub for it and replace with `yq-go`. See https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/monitoringpvc/ipi-install-monitoringpvc-commands.sh for example

Reply:
Get it, will update it.